iModelCore/ECDb/ECDb/ECSql/ECSqlPreparer.cpp

```diff
@@ -104,19 +104,43 @@ ECSqlStatus ECSqlExpPreparer::PrepareAllOrAnyExp(ECSqlPrepareContext& ctx, AllOr
 
     ctx.GetSqlBuilder().Append("EXISTS").AppendParenLeft();
 
-    SelectStatementExp const* subquery = (exp.GetSubquery())->GetQuery();
-    ECSqlStatus stat = ECSqlSelectPreparer::Prepare(ctx, *subquery);
-    if (stat != ECSqlStatus::Success)
-        return stat;
+    SelectStatementExp const* selectSubquery = (exp.GetSubquery())->GetQuery<SelectStatementExp>();
+    CommonTableExp const* cteSubquery = (exp.GetSubquery())->GetQuery<CommonTableExp>();

```

Comment:
It would be cleaner to get the CTE right above where you check for it to be null instead of here.

Reply:
fixed